libs/pavexc/src/compiler/analyses/framework_items.rs

```diff
@@ -105,6 +105,21 @@ impl FrameworkItemDb {
                 binding: format_ident!("allowed_methods"),
             },
         );
+
+        let connection_info = process_framework_path(
+            "pavex::connection::ConnectionInfo",
+            package_graph,
+            krate_collection,
+        );
+        items.insert(connection_info, Self::connection_info());
+        id2metadata.insert(
+            Self::connection_info(),
+            FrameworkItemMetadata {
+                lifecycle: Lifecycle::Singleton,
```

Comment:
This should be request scoped.

Reply:
https://github.com/LukeMathWalker/pavex/pull/169/commits/8c95643c685b929a251e48b6af37bbf4bcc7e680